app/controllers/welcome_controller.rb

```diff
@@ -3,17 +3,15 @@ class WelcomeController < ApplicationController
   before_action :set_cookie_attributes, only: [:index]
 
   def show_hints
+    authorize current_user, :view?
```

Comment:
We are following the pattern, so you might need to correct and `authorize current_user`, and add a `show_hints` method in the `user_policy.rb` file.

```
def show_hints?
  view?
end
```